ansible/library/topo_facts.py

```diff
@@ -18,19 +19,90 @@
       required: True
 '''
 
+def get_vm_type(ngs_type):
+    if ngs_type == "ToRRouter":
+        return "tor"
+    if ngs_type == "LeafRouter":
+        return "leaf"
+    if ngs_type == "SpineRouter":
+        return "spine"
+
 class ParseTestbedTopoinfo():
     '''
     Parse topology yml file
     '''
     def __init__(self):
         self.vm_topo_config = {}
 
+    def parse_clet(self, lst, cfg, fpath):
+        # Get the neighbors list
+        for e in lst:
+            if e.has_key('DEVICE_NEIGHBOR_METADATA'):
+                for vm in e['DEVICE_NEIGHBOR_METADATA']:
+                    cfg[vm] = dict()
+                    cfg[vm]['intfs'] = []
+                    cfg[vm]['properties'] = ["common"]
+                    cfg[vm]['properties'].append(get_vm_type(e['DEVICE_NEIGHBOR_METADATA'][vm]["type"]))
+
+        # fill data for each VM
+        for e in lst:
+            if e.has_key('DEVICE_NEIGHBOR'):
+                for intf in e['DEVICE_NEIGHBOR']:
+                    vm = e['DEVICE_NEIGHBOR'][intf]['name']
+                    cfg[vm]['interface_indexes'] = [int(intf[-2:]), ]
```

Comment:
Vlan index could be incorrect if extracted from interface name. As discussed, please calculate the index from the topo yml

Reply:
Fixed